src/asio_service.cxx

```diff
@@ -1458,8 +1458,13 @@ void _timer_handler_(ptr<delayed_task>& task, ERROR_CODE err) {
 asio_service_impl::asio_service_impl(const asio_service::options& _opt,
                                      ptr<logger> l)
     : io_svc_()
-    , ssl_server_ctx_(ssl_context::sslv23)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
+    , ssl_server_ctx_(ssl_context::tlsv12_client)
+    , ssl_client_ctx_(ssl_context::tlsv12_server)
```

Comment:
Seems to me `tlsv12_client` and `tlsv12_server` are swapped. Could you please correct them? Thanks.

Reply:
thanks a lot. fixed now.